Nightly failures builds: fatal error: Cuda/Kokkos_Cuda_Half.hpp: No such file or directory

Nightly builds of KokkosKernels and Trilinos began failing 1/21/2022 with output:

```
22:07:04 /data/jenkins-new/workspace/KokkosKernels_semsrhel7gpu01_cuda111_gcc830-dbg-boundscheck/kokkos-kernels/unit_test/sparse/Test_Sparse_spmv.hpp:13:10: fatal error: Cuda/Kokkos_Cuda_Half.hpp: No such file or directory
22:07:04  #include "Cuda/Kokkos_Cuda_Half.hpp"
```

Possibly due to changes with kokkos/kokkos#4650, @e10harvey are you able to investigate?

**Sample reproducer** (cuda build):
```
module load sems-archive-env sems-archive-cmake/3.17.1 sems-archive-gcc/8.3.0 sems-archive-cuda/11.1

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=Cuda,Pthread --arch=Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --with-cuda-options=enable_lambda --boundscheck  --no-examples  --debug
```

--- a/unit_test/sparse/Test_Sparse_spmv.hpp
+++ b/unit_test/sparse/Test_Sparse_spmv.hpp
@@ -10,7 +10,6 @@
 
 #include "KokkosKernels_Controls.hpp"
 #include "KokkosKernels_default_types.hpp"
-#include "Cuda/Kokkos_Cuda_Half.hpp"
 
 // #ifndef kokkos_complex_double
 // #define kokkos_complex_double Kokkos::complex<double>

